validator/slashing-protection/local/attester_protect.go

```diff
@@ -0,0 +1,180 @@
+package local
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/sirupsen/logrus"
+
+	"github.com/prysmaticlabs/prysm/shared/mputil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/validator/db/kv"
+	"github.com/prysmaticlabs/prysm/validator/slashing-protection"
+)
+
+// IsSlashableAttestation determines if an incoming attestation is slashable
+// according to local protection. Then, if the attestation successfully passes
+// checks, we update our local attesting history accordingly.
+func (s *Service) IsSlashableAttestation(
+	ctx context.Context,
+	indexedAtt *ethpb.IndexedAttestation,
+	pubKey [48]byte,
+	signingRoot [32]byte,
+) (bool, error) {
+	if indexedAtt == nil || indexedAtt.Data == nil {
+		return false, errors.New("received nil attestation")
+	}
+	lock := mputil.NewMultilock(fmt.Sprintf("%x", pubKey))
```

Comment:
Do you have to format it like this? A string is basically just a byte slice so `string(pubKey)` would work the same in terms of uniqueness but be much faster than any formatting.